### ISSUE TITLE
Colorful output on terminal

### ISSUE THREAD

**Issue by nimitbhardwaj** [chapel-lang/chapel#8321]
Just want that there are some functions which can write colourfully  on terminal console. Because mostly all terminals support [ASCII](https://stackoverflow.com/questions/3219393/stdlib-and-colored-output-in-c) schemes, so it can be done easily, just I tried this
```chapel
writeln("\x1b[34m Hello, world!\x1b[0m");
```

This wrote `Hello World` in blue, so I wanted something like this 
`writeBlueln("Hello");`, this will write Hello in Blue, similarly for other colors, and also there is an alternative, `writeln("%sB Hello %eB");` , this will write again Hello in blue, but here %sB is tag for starting of blue, and %eB is for ending of blue, I think both types of functions can be a good layer on writeln and other writing functions of type.

How about implementing it, an easy think to implement can be useful often for command line applications, there are many such useful modules in python too, like colorama etc. So It can be useful.

**Comment by ben-albrecht** [comment]
I think you could pursue this as a [mason package](https://chapel-lang.org/docs/latest/tools/mason/mason.html) in a separate repository.

Here is an [example](https://gist.github.com/ben-albrecht/d876460d8dacdaa16e3d336036d80022) I made a while ago, if that helps.

**Comment by ben-albrecht** [comment]
To be clear, I think this would be a nice library to have as a mason package.

However, I am going to close this issue since it is not (yet) relevant to the chapel-lang/chapel repository.

I suggest you create a separate repository for this library and create an issue to discuss the design there. You can request feedback from the developer community by emailing the chapel-developers mailing list.

**Comment by ben-albrecht** [comment]
I am reopening this to give others a chance for opinions...

**Comment by bradcray** [comment]
I like the Colors.chpl sketch that Ben points to above as a means of switching colors within a single write() call in a clear manner.  I worry a little about adding more control strings into `writef()` due to the number that already exist there, at least without a clear precedent in a popular language that has gained a foothold (I'm not aware of one).

**Comment by nimitbhardwaj** [comment]
OK, what will be better, making the new module, include in mason, or edit the writef, write, writeln etc functions to support colors in some easy way then to write those weird ANSI codes.

**Comment by nimitbhardwaj** [comment]
Yes, I feel that it would be great for Mason, implementing for standard writing functions will be mess here, so implementing using a new package in Mason will likely to be best.

**Comment by nimitbhardwaj** [comment]
Hey, for console output of colors I made the repo [here](https://github.com/nimitbhardwaj/ChromaTerminos). Please have a check there.

**Comment by nimitbhardwaj** [comment]
And made the PR at the mason-registry too for the same
